File: example/src/main/java/com/stripe/example/controller/ErrorDialogHandler.java
package com.stripe.example.controller;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;

import com.stripe.example.R;
import com.stripe.example.dialog.ErrorDialogFragment;

import java.lang.ref.WeakReference;

/**
 * A convenience class to handle displaying error dialogs.
 */
public class ErrorDialogHandler {

    @NonNull private final WeakReference<AppCompatActivity> mActivityRef;

    public ErrorDialogHandler(@NonNull AppCompatActivity activity) {
        mActivityRef = new WeakReference<>(activity);
    }

    public void show(@NonNull String errorMessage) {
        final AppCompatActivity activity = mActivityRef.get();
        if (activity == null) {
            return;
        }

        ErrorDialogFragment.newInstance(activity.getString(R.string.validationErrors), errorMessage)
                .show(activity.getSupportFragmentManager(), "error");
    }
}

File: example/src/main/java/com/stripe/example/controller/ErrorDialogHandler.kt
package com.stripe.example.controller

import androidx.appcompat.app.AppCompatActivity

import com.stripe.example.R
import com.stripe.example.dialog.ErrorDialogFragment

import java.lang.ref.WeakReference

/**
 * A convenience class to handle displaying error dialogs.
 */
class ErrorDialogHandler(activity: AppCompatActivity) {

    private val activityRef: WeakReference<AppCompatActivity> = WeakReference(activity)

    fun show(errorMessage: String) {
        val activity = activityRef.get() ?: return

        ErrorDialogFragment.newInstance(activity.getString(R.string.validationErrors), errorMessage)
            .show(activity.supportFragmentManager, "error")
    }
}
